Formula/inlets.rb

```diff
@@ -0,0 +1,37 @@
+class Inlets < Formula
+  desc "Expose your local endpoints to the Internet"
+  homepage "https://github.com/alexellis/inlets"
+  url "https://github.com/alexellis/inlets.git",
+      :tag      => "0.5.7",
+      :revision => "4c5a519b669cfb92d3ced715fc8e82489aded46a"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["XC_ARCH"] = "amd64"
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/alexellis/inlets").install buildpath.children
+    cd "src/github.com/alexellis/inlets" do
+      commit = Utils.popen_read("git", "rev-parse", "HEAD").chomp
+      system "go", "build", "-ldflags",
+             "-s",
+             "-w",
+             "-X main.GitCommit=#{commit}",
+             "-X main.Version=#{version}",
+             "-a",
+             "-installsuffix", "cgo", "-o", bin/"inlets"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    begin
+      stable_resource = stable.instance_variable_get(:@resource)
+      commit = stable_resource.instance_variable_get(:@specs)[:revision]
+      inlets_version = shell_output("#{bin}/inlets -version")
+      assert_match /\s#{commit}$/, inlets_version
+      assert_match /\s#{version}$/, inlets_version
```

Comment:
We need a test that exercises the some of the functionality of the app, not just print the version number.

Reply:
This is already tested upstream. Do you want me to just convert this to pull a binary?

Reply:
As per faas-cli every version is tested upstream before being bumped in the recipe. I don't want to have the burden of writing and maintaining tests in two languages.

Reply:
- Homebrew core formulas must build from source.
- Functional tests are important so we don't ship broken binaries. Yes, you do test upstream, but sometimes the build or bottling system here in Homebrew can cause trouble, which is why we want to ensure there are checks in place.
- Otherwise, distribute the software in your own tap, where you have full control of what you do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

Reply:
Example without detailed tests: docker.rb, vim.rb, (etc, etc, etc)

Integrating testing this tool e2e which uses web sockets will be extremely challenging.

If you or one of your maintainers would like to write them I'd be happy to review them, but this seems particularly burdensome. I'm not sure why this formula should be different to the others?

Reply:
> Each formula is reviewed individually.

> There's also the possibility that formula predates some changes that haven't been retroactively applied.

A lot of tests were added because they were better than nothing on existing formulae. Ideally we'd have extensive tests everywhere but until that is the case we'll have to accept just keeping new entries to this standard.

I get that writing a test might be difficult but this "they did it too"-attitude isn't helping anyone.

Reply:
I can't see anything in the PR template or the contributing guide that suggests that I need to write e2e integration tests for formulas. https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md

Writing a test for this in Ruby in a formula format is going to be incredibly time-consuming and for little gain. I have no idea where to start since this isn't just a quick HTTP call like with faas-cli. This is a CLI which can act as client or server to form a HTTP tunnel over websockets.

Help me out here. I've done all the steps in the contribution guide, fixed up the stylistic requests, but what you're asking for is extremely hard to achieve without expert brew knowledge and experience. I can't dedicate 1-2 weeks to writing an e2e test for the Formula.

Please tell me what you need to get this Formula merged for our users.